clcache.py

```diff
@@ -1529,25 +1480,25 @@ def processCompileRequest(cache, compiler, args):
     try:
         sourceFiles, objectFiles = CommandLineAnalyzer.analyze(cmdLine)
 
-        if len(sourceFiles) > 1:
-            return reinvokePerSourceFile(cmdLine, sourceFiles, environment), '', ''
-        else:
-            assert objectFile is not None
-            if 'CLCACHE_NODIRECT' in os.environ:
-                returnCode, compilerOutput, compilerStderr, cleanupRequired = \
-                    processNoDirect(cache, objectFiles[0], compiler, cmdLine, environment)
-            else:
-                returnCode, compilerOutput, compilerStderr, cleanupRequired = \
-                    processDirect(cache, objectFiles[0], compiler, cmdLine, sourceFiles[0])
-            printTraceStatement("Finished. Exit code {0:d}".format(returnCode))
-
-            if cleanupRequired:
-                with cache.lock:
-                    cleanCache(cache)
+        baseCmdLine = []
+        for arg in cmdLine:
+            if arg not in sourceFiles:
+                baseCmdLine.append(arg)
+        with concurrent.futures.ThreadPoolExecutor(max_workers=jobCount(cmdLine)) as executor:
+            jobs = []
+            for i,srcFile in enumerate(sourceFiles):
+                jobCmdLine = list(baseCmdLine)
+                jobCmdLine.append(srcFile)
+                jobs.append(executor.submit(processActualCompileRequest, cache, compiler, jobCmdLine, srcFile, objectFiles[i], environment))
```

Comment:
Instead of scheduling a `processActualCompileRequest` here, which then uses most (but not all!) arguments depending on whether `CLCACHE_NODIRECT` is set, how about testing `CLCACHE_NODIRECT` just once up front here and then either scheduling a call to `processDirect` (passing the environment) or `processNoDirect` (passing the source file)?

Also, I guess it might work to factor out the final (optional) cache cleaning step so that it's performed just once at the end of all compiles?